### ISSUE TITLE
appName.toLowerCase is not a function issue

### ISSUE THREAD

**Issue by n1k0nx** [balena-io/balena-cli#1187]
- **Cli version: latest (24/04/2019)**
- **Standalone install:** yes/no
- **Node version: balenaOS 2.31.5+rev1**
- **Npm version: latest (24/04/2019)**
- **Operating system and architecture: tryed on windows 10 and ubuntu 64bit**
hello, sorry I'm a niubbo on Balena app :) I would like to create Balena sense device. I've tryed many ways and many commands on cli but I receive always the red error on cli "appname.tolowecase is not a function" message. could you help me please ? 
thank you and regards
n1k0nx
p.s. sorry for my english
---

*Please keep in mind that we try to use the issue tracker of this repository for specific bug reports & CLI feature requests. General & troubleshooting questions are encouraged to be posted to the [balena forums](https://forums.balena.io) where the community can both contribute and benefit from the answers.*

*Before submitting this issue please check that this issue is not a duplicate. If there is another issue describing the same problem or feature please add your information to the existing issue's comments.*

**Comment by pdcastro** [comment]
@n1k0nx, please run the CLI in debug mode by setting a DEBUG=1 environment variable on the same terminal window where you are running the CLI:
* Windows command prompt: `set DEBUG=1`
* Linux: `export DEBUG=1`

Then please share an exact copy-and-paste of:
* The command line you are executing, for example: `balena push balenaSense`
* The full error message in debug mode.
* The output of running "balena version". I appreciate you mentioned it is the "latest" version but we've had 4 or 5 releases in the past 24 hours! (10.2.0, 10.3.0, 10.4.0, 10.4.1, 10.5.0...)

Also, is the `balena login` command successful? Thanks!

**Comment by n1k0nx** [comment]
Hi Paulo
Thank you for quicly reply. I used on win10 Balena CLI 10.1.1 and on Ubuntu 10.4.0 now I'm downloading 10.6.0. 

Could you kindly specify how to set DEBUG=1 ? command ? I use like this?
balena set DEBUG=1 or something else? 

p.s. I can login on cli in win10 and ubuntu (virtual machine)

thank you!
luca

**Comment by pdcastro** [comment]
@n1k0nx, the sequence of commands on a Windows 10 command prompt would be:
```
Z:\balena-sense-master> set DEBUG=1
Z:\balena-sense-master> balena version
10.6.0
Z:\balena-sense-master> balena push balenaSense
...
```

**Comment by n1k0nx** [comment]
Hi Paulo
nothing to do. 
I've done your istructions but I get this message 

C:\projects\balena-sense-master>set DEBUG=1

C:\projects\balena-sense-master>balena push 1234 balenaSense
Usage: balena [COMMAND] [OPTIONS]

If you need help, or just want to say hi, don't hesitate in reaching out
through our discussion and support forums at https://forums.balena.io

For bug reports or feature requests, have a look at the GitHub issues or
create a new one at: https://github.com/balena-io/balena-cli/issues/

Primary commands:

    app <name>                          list a single application
    apps                                list all applications
    build [source]                      Build a single image or a multicontainer project locally
    deploy <appName> [image]            Deploy a single image or a multicontainer project to a balena application
    device <uuid>                       list a single device
    devices                             list all devices
    help [command...]                   show help
    join [deviceIp]                     Promote a local device running balenaOS to join an application on a balena server
    leave [deviceIp]                    Detach a local device from its balena application
    local scan                          Scan for balenaOS devices in your local network
    login                               login to balena
    logs <uuidOrDevice>                 show device logs
    preload <image>                     preload an app on a disk image (or Edison zip archive)
    push <applicationOrDevice>          Start a remote build on the balena cloud build servers or a local mode device
    ssh [uuid]                          get a shell into the running app container of a device
    tunnel <uuid>                       Tunnel local ports to your balenaOS device

Run `balena help --verbose` to list additional commands

Global Options:

    --help, -h

C:\projects\balena-sense-master>


_________________________________________________________________________________________


the 1234 is the name of my application (starter) 
thank you in advance 
best regards
luca

**Comment by pdcastro** [comment]
@n1k0nx, the word `balenaSense` in the [project blog](https://www.balena.io/blog/build-an-environment-and-air-quality-monitor-with-raspberry-pi/) is just a sample application name. If your app name is 1234, then the command is `balena push 1234`:
```
C:\projects\balena-sense-master> set DEBUG=1
C:\projects\balena-sense-master> balena push 1234
```

**Comment by pdcastro** [comment]
@n1k0nx, thanks for the debug output. I know what's happening: the CLI framework we currently use (Capitano) automatically converts any number-looking argument from string to number, so the `1234` app name was incorrectly converted to a number -- and a number cannot be further converted to lower case letters ("toLowerCase is not a function" on a numeric argument).

The immediate workaround is to rename the app or create a new one, with a name that includes at least one non-numeric character, for example "a1234". I am working on a bug fix and with a bit of luck, there will be a fixed CLI release in the next 24 hours.

Thank you for reporting this issue!

**Comment by n1k0nx** [comment]
👍 thank you Paulo! I'm pleased to be useful  :) 

regards
luca

[Info]      Build finished in 3 minutes, 29 seconds
[debug] handling message: {"message":"\u001b[1m\u001b[34m\t\t\t    \\\n\t\t\t     \\\n\t\t\t      \\\\\n\t\t\t       \\\\\n\t\t\t        >\\/7\n\t\t\t    _.-(6'  \\\n\t\t\t   (=___._/` \\\n\t\t\t        )  \\ |\n\t\t\t       /   / |\n\t\t\t      /    > /\n\t\t\t     j    < _\\\n\t\t\t _.-' :      ``.\n\t\t\t \\ r=._\\        `.\n\t\t\t<`\\\\_  \\         .`-.\n\t\t\t \\ r-7  `-. ._  ' .  `\\\n\t\t\t  \\`,      `-.`7  7)   )\n\t\t\t   \\/         \\|  \\'  / `-._\n\t\t\t              ||    .'\n\t\t\t               \\\\  (\n\t\t\t                >\\  >\n\t\t\t            ,.-' >.'\n\t\t\t           <.'_.''\n\t\t\t             <'\u001b[39m\u001b[22m","isSuccess":true}
                            \
                             \
                              \\
                               \\
                                >\/7
                            _.-(6'  \
                           (=___._/` \
                                )  \ |
                               /   / |
                              /    > /
                             j    < _\
                         _.-' :      ``.
                         \ r=._\        `.
                        <`\\_  \         .`-.
                         \ r-7  `-. ._  ' .  `\
                          \`,      `-.`7  7)   )
                           \/         \|  \'  / `-._
                                      ||    .'
                                       \\  (
                                        >\  >
                                    ,.-' >.'
                                   <.'_.''
                                     <'

**Comment by pdcastro** [comment]
This issue was fixed in CLI version 10.8.2.

**Comment by n1k0nx** [comment]
Thank you Paulo!
